samples/sample_multi_transcode/src/sample_multi_transcode.cpp

```diff
@@ -829,31 +844,31 @@ mfxStatus Launcher::CreateSafetyBuffers()
 
 void Launcher::Close()
 {
-    while(m_pSessionArray.size())
+    while(m_pThreadContextArray.size())
     {
-        delete m_pSessionArray[m_pSessionArray.size()-1];
-        m_pSessionArray[m_pSessionArray.size() - 1] = NULL;
-        m_pSessionArray.pop_back();
+        delete m_pThreadContextArray[m_pThreadContextArray.size()-1];
+        m_pThreadContextArray[m_pThreadContextArray.size() - 1] = nullptr;
```

Comment:
1 .Whether it makes sense when it's followed by
`m_pThreadContextArray.pop_back(); `
?
2. I'd rather use
`m_pThreadContextArray.back()`
instead of 
`m_pThreadContextArray[m_pThreadContextArray.size()-1]`
and
`while (!m_pThreadContextArray.empty())`
instead of
`while(m_pThreadContextArray.size())`

BTW, functional approach might be even better -
```
std::for_each(std::rbegin(m_pThreadContextArray), std::rend(m_pThreadContextArray), std::default_delete<ThreadTranscodeContext>{});
m_pThreadContextArray.clear();
```

Reply:
absolutely agree